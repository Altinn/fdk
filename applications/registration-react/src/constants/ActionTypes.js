export const PROVENANCE_REQUEST = 'PROVENANCE_REQUEST';
export const PROVENANCE_SUCCESS = 'PROVENANCE_SUCCESS';
export const PROVENANCE_FAILURE = 'PROVENANCE_FAILURE';

export const FREQUENCY_REQUEST = 'FREQUENCY_REQUEST';
export const FREQUENCY_SUCCESS = 'FREQUENCY_SUCCESS';
export const FREQUENCY_FAILURE = 'FREQUENCY_FAILURE';

export const REFERENCEDATASETS_REQUEST = 'REFERENCEDATASETS_REQUEST';
export const REFERENCEDATASETS_SUCCESS = 'REFERENCEDATASETS_SUCCESS';
export const REFERENCEDATASETS_FAILURE = 'REFERENCEDATASETS_FAILURE';

export const USER_REQUEST = 'USER_REQUEST';
export const USER_SUCCESS = 'USER_SUCCESS';
export const USER_FAILURE = 'USER_FAILURE';

export const CATALOGS_REQUEST = 'CATALOGS_REQUEST';
export const CATALOGS_SUCCESS = 'CATALOGS_SUCCESS';
export const CATALOGS_FAILURE = 'CATALOGS_FAILURE';

export const PUBLISHDATASET = 'PUBLISHDATASET';

export const DATASET_LAST_SAVED = 'DATASET_LAST_SAVED';
