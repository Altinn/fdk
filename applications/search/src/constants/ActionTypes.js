export const DATASETS_REQUEST = 'DATASETS_REQUEST';
export const DATASETS_SUCCESS = 'DATASETS_SUCCESS';
export const DATASETS_FAILURE = 'DATASETS_FAILURE';

export const TERMS_REQUEST = 'TERMS_REQUEST';
export const TERMS_SUCCESS = 'TERMS_SUCCESS';
export const TERMS_FAILURE = 'TERMS_FAILURE';

export const THEMES_REQUEST = 'THEMES_REQUEST';
export const THEMES_SUCCESS = 'THEMES_SUCCESS';
export const THEMES_FAILURE = 'THEMES_FAILURE';

export const PUBLISHERS_REQUEST = 'PUBLISHERS_REQUEST';
export const PUBLISHERS_SUCCESS = 'PUBLISHERS_SUCCESS';
export const PUBLISHERS_FAILURE = 'PUBLISHERS_FAILURE';
