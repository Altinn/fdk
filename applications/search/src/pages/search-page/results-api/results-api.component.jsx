import React from 'react';
import PropTypes from 'prop-types';
import ReactPaginate from 'react-paginate';
import { Button, Modal, ModalBody, ModalFooter, ModalHeader } from 'reactstrap';
import cx from 'classnames';
import _ from 'lodash';
import { withRouter } from 'react-router';

import localization from '../../../lib/localization';
import { SearchHitItem } from './search-hit-item/search-hit-item.component';
import { FilterBox } from '../../../components/filter-box/filter-box.component';
import { SearchPublishersTree } from '../search-publishers-tree/search-publishers-tree.component';
import { getSortfield, setSortfield } from '../search-location-helper';

const renderFilterModal = ({
  showFilterModal,
  closeFilterModal,
  apiAggregations,
  locationSearch,
  publisherCounts,
  publishers,
  onFilterFormat,
  onFilterPublisherHierarchy
}) => (
  <Modal isOpen={showFilterModal} toggle={closeFilterModal}>
    <ModalHeader toggle={closeFilterModal}>{localization.filter}</ModalHeader>
    <ModalBody>
      <div className="search-filters">
        <SearchPublishersTree
          title={localization.facet.provider}
          publisherCounts={publisherCounts}
          onFilterPublisherHierarchy={onFilterPublisherHierarchy}
          activeFilter={locationSearch.orgPath}
          publishers={publishers}
        />
        <FilterBox
          htmlKey={2}
          title={localization.facet.format}
          filter={_.get(apiAggregations, 'formats')}
          onClick={onFilterFormat}
          activeFilter={locationSearch.format}
        />
      </div>
    </ModalBody>
    <ModalFooter>
      <Button className="fdk-button" onClick={closeFilterModal} color="primary">
        {localization.close}
      </Button>
    </ModalFooter>
  </Modal>
);

const renderHits = (hits, publishers) => {
  if (hits && Array.isArray(hits)) {
    return hits.map((item, index) => (
      <SearchHitItem
        key={item.id}
        item={item}
        fadeInCounter={index < 3 ? index : null}
        publishers={publishers}
      />
    ));
  }
  return null;
};

export const ResultsApiPure = ({
  showFilterModal,
  closeFilterModal,
  apiItems,
  apiTotal,
  apiAggregations,
  onFilterPublisherHierarchy,
  onFilterFormat,
  locationSearch,
  publisherCounts,
  publishers,
  onClearFilters,
  onPageChange,
  showClearFilterButton,
  hitsPerPage,
  history,
  location
}) => {
  const page = parseInt(locationSearch.page || 0, 10);
  const pageCount = Math.ceil((apiTotal || 1) / hitsPerPage);

  const clearButtonClass = cx(
    'btn',
    'btn-primary',
    'fdk-button',
    'fade-in-500',
    {
      'd-none': !showClearFilterButton
    }
  );

  const sortfield = getSortfield(location);
  const sortByScoreClass = cx('fdk-button', 'fdk-button-black-toggle', {
    selected: !sortfield
  });
  const sortByLastModifiedClass = cx('fdk-button', 'fdk-button-black-toggle', {
    selected: sortfield === 'modified'
  });

  const onSortByScoreClick = () => {
    setSortfield(history, location, undefined);
  };
  const onSortByModifiedClick = () => {
    setSortfield(history, location, 'modified');
  };

  return (
    <main data-test-id="apis" id="content">
      <div className="row mb-3">
        <div className="col-6 col-lg-4">
          <button
            className={clearButtonClass}
            onClick={onClearFilters}
            type="button"
          >
            {localization.query.clear}
          </button>
        </div>
        <div className="col-6 col-lg-4 offset-lg-4">
          <div className="d-flex justify-content-end">
            <Button
              className={sortByScoreClass}
              onClick={onSortByScoreClick}
              color="primary"
            >
              {localization.sort.relevance}
            </Button>
            <Button
              className={sortByLastModifiedClass}
              onClick={onSortByModifiedClick}
              color="primary"
            >
              {localization.sort.modified}
            </Button>
          </div>
        </div>
      </div>
      <div className="row">
        <aside className="search-filters col-lg-4 d-none d-lg-block">
          <span className="uu-invisible" aria-hidden="false">
            Filtrering
          </span>
          {apiAggregations && (
            <div>
              {renderFilterModal({
                showFilterModal,
                closeFilterModal,
                apiAggregations,
                locationSearch,
                publisherCounts,
                publishers,
                onFilterFormat,
                onFilterPublisherHierarchy
              })}
              <SearchPublishersTree
                title={localization.facet.provider}
                publisherCounts={publisherCounts}
                onFilterPublisherHierarchy={onFilterPublisherHierarchy}
                activeFilter={locationSearch.orgPath}
                publishers={publishers}
              />
              <FilterBox
                htmlKey={2}
                title={localization.facet.format}
                filter={_.get(apiAggregations, 'formats')}
                onClick={onFilterFormat}
                activeFilter={locationSearch.format}
              />
            </div>
          )}
        </aside>
        <div id="apis" className="col-12 col-lg-8">
          {renderHits(apiItems, publishers)}

          <div className="col-12 d-flex justify-content-center">
            <span className="uu-invisible" aria-hidden="false">
              Sidepaginering.
            </span>
            <ReactPaginate
              pageCount={pageCount}
              pageRangeDisplayed={2}
              marginPagesDisplayed={1}
              previousLabel={localization.page.prev}
              nextLabel={localization.page.next}
              breakLabel={<span>...</span>}
              breakClassName="break-me"
              containerClassName="pagination"
              onPageChange={onPageChange}
              subContainerClassName="pages pagination"
              activeClassName="active"
              initialPage={page}
              disableInitialCallback
            />
          </div>
        </div>
      </div>
    </main>
  );
};

ResultsApiPure.defaultProps = {
  showFilterModal: false,
  closeFilterModal: _.noop,
  showClearFilterButton: false,

  apiItems: [],
  apiTotal: 0,
  apiAggregations: null,

  onFilterPublisherHierarchy: _.noop,
  onFilterFormat: _.noop,
  onClearFilters: _.noop,
  locationSearch: {},
  publisherCounts: null,
  publishers: null,

  onPageChange: _.noop,
  hitsPerPage: 0,

  history: { push: _.noop },
  location: { search: '' }
};

ResultsApiPure.propTypes = {
  showFilterModal: PropTypes.bool,
  closeFilterModal: PropTypes.func,
  showClearFilterButton: PropTypes.bool,

  apiItems: PropTypes.array,
  apiTotal: PropTypes.number,
  apiAggregations: PropTypes.object,

  onFilterPublisherHierarchy: PropTypes.func,
  onFilterFormat: PropTypes.func,
  onClearFilters: PropTypes.func,
  locationSearch: PropTypes.object,

  publisherCounts: PropTypes.array,
  publishers: PropTypes.object,

  onPageChange: PropTypes.func,
  hitsPerPage: PropTypes.number,

  history: PropTypes.object,
  location: PropTypes.object
};

export const ResultsApi = withRouter(ResultsApiPure);
