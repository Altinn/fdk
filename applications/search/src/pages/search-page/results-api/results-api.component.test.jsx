import React from 'react';
import { shallow } from 'enzyme';
import _ from 'lodash';
import { ResultsApiPure } from './results-api.component';
import apisApiResponse from '../../../mock/apis.response.json';
import { HITS_PER_PAGE } from '../../../constants/constants';

let closeFilterModal;
let onFilterTheme;
let onFilterAccessRights;
let onFilterPublisherHierarchy;
let onClearFilters;
let setApiSort;
let onSortByLastModified;
let onSortByScore;
let onPageChange;
let defaultProps;
let wrapper;

beforeEach(() => {
  closeFilterModal = jest.fn();
  onFilterTheme = jest.fn();
  onFilterAccessRights = jest.fn();
  onFilterPublisherHierarchy = jest.fn();
  onClearFilters = jest.fn();
  setApiSort = jest.fn();
  onSortByLastModified = jest.fn();
  onSortByScore = jest.fn();
  onPageChange = jest.fn();

  defaultProps = {
    showFilterModal: true,
    apiItems: {},
    searchQuery: {},
    themesItems: {},
    publisherArray: [],
    publishers: {},
    showClearFilterButton: false,
    hitsPerPage: HITS_PER_PAGE,
    closeFilterModal,
    onFilterTheme,
    onFilterAccessRights,
    onFilterPublisherHierarchy,
    onClearFilters,
    setApiSort,
    onSortByLastModified,
    onSortByScore,
    onPageChange
  };
  wrapper = shallow(<ResultsApiPure {...defaultProps} />);
});

test('should render ResultsApi correctly with minimum of props and no hits', () => {
  const minWrapper = shallow(
    <ResultsApiPure
      setApiSort={setApiSort}
      onSortByLastModified={onSortByLastModified}
      onSortByScore={onSortByScore}
    />
  );
  expect(minWrapper).toMatchSnapshot();
});

test('should render ResultsApi correctly with hits', () => {
  wrapper.setProps({
    apiItems: _.get(apisApiResponse, 'hits'),
    apiAggregations: _.get(apisApiResponse, 'aggregations'),
    apiTotal: _.get(apisApiResponse, 'total')
  });
  expect(wrapper).toMatchSnapshot();
});
