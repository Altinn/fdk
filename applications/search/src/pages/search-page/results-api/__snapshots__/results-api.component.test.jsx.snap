// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ResultsApi correctly with hits 1`] = `
<main
  data-test-id="apis"
  id="content"
>
  <div
    className="row mb-3"
  >
    <div
      className="col-6 col-lg-4"
    >
      <button
        className="btn btn-primary fdk-button fade-in-500 d-none"
        onClick={[MockFunction]}
        type="button"
      >
        Clear
      </button>
    </div>
    <div
      className="col-6 col-lg-4 offset-lg-4"
    >
      <div
        className="d-flex justify-content-end"
      >
        <Button
          className="fdk-button fdk-button-black-toggle selected"
          color="primary"
          onClick={[Function]}
          tag="button"
        >
          Relevance
        </Button>
        <Button
          className="fdk-button fdk-button-black-toggle"
          color="primary"
          onClick={[Function]}
          tag="button"
        >
          Recently published
        </Button>
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <aside
      className="search-filters col-lg-4 d-none d-lg-block"
    >
      <span
        aria-hidden="false"
        className="uu-invisible"
      >
        Filtrering
      </span>
      <div>
        <Modal
          autoFocus={true}
          backdrop={true}
          backdropTransition={
            Object {
              "mountOnEnter": true,
              "timeout": 150,
            }
          }
          centered={false}
          fade={true}
          isOpen={true}
          keyboard={true}
          modalTransition={
            Object {
              "timeout": 300,
            }
          }
          onClosed={[Function]}
          onOpened={[Function]}
          role="dialog"
          toggle={[MockFunction]}
          zIndex={1050}
        >
          <ModalHeader
            closeAriaLabel="Close"
            tag="h5"
            toggle={[MockFunction]}
            wrapTag="div"
          >
            Filter
          </ModalHeader>
          <ModalBody
            tag="div"
          >
            <div
              className="search-filters"
            >
              <SearchPublishersTree
                activeFilter={null}
                filter={Array []}
                onFilterPublisherHierarchy={[MockFunction]}
                onSearch={null}
                orgPath={null}
                publishers={Object {}}
                title="Provider"
              />
              <FilterBox
                activeFilter={null}
                filter={
                  Object {
                    "buckets": Array [
                      Object {
                        "count": 1,
                        "key": "MISSING",
                      },
                      Object {
                        "count": 1,
                        "key": "application/json",
                      },
                      Object {
                        "count": 1,
                        "key": "application/xml",
                      },
                      Object {
                        "count": 1,
                        "key": "text/json",
                      },
                      Object {
                        "count": 1,
                        "key": "text/xml",
                      },
                    ],
                  }
                }
                htmlKey={2}
                onClick={null}
                themesItems={null}
                title="Format"
              />
            </div>
          </ModalBody>
          <ModalFooter
            tag="div"
          >
            <Button
              className="fdk-button"
              color="primary"
              onClick={[MockFunction]}
              tag="button"
            >
              Close
            </Button>
          </ModalFooter>
        </Modal>
        <SearchPublishersTree
          activeFilter={null}
          filter={Array []}
          onFilterPublisherHierarchy={[MockFunction]}
          onSearch={null}
          orgPath={null}
          publishers={Object {}}
          title="Provider"
        />
        <FilterBox
          activeFilter={null}
          filter={
            Object {
              "buckets": Array [
                Object {
                  "count": 1,
                  "key": "MISSING",
                },
                Object {
                  "count": 1,
                  "key": "application/json",
                },
                Object {
                  "count": 1,
                  "key": "application/xml",
                },
                Object {
                  "count": 1,
                  "key": "text/json",
                },
                Object {
                  "count": 1,
                  "key": "text/xml",
                },
              ],
            }
          }
          htmlKey={2}
          onClick={null}
          themesItems={null}
          title="Format"
        />
      </div>
    </aside>
    <div
      className="col-12 col-lg-8"
      id="apis"
    >
      <SearchHitItem
        fadeInCounter={0}
        item={
          Object {
            "formats": Array [],
            "id": "68dda78f-364f-41dd-869a-0dd8d9a3fe4c",
            "nationalComponent": false,
            "publisher": Object {
              "id": "970018131",
              "name": "UTDANNINGSDIREKTORATET",
              "orgPath": "/STAT/872417842/970018131",
              "prefLabel": Object {
                "no": "Utdanningsdirektoratet",
              },
            },
            "title": "Barnehagefakta",
            "titleFormatted": "Barnehagefakta",
          }
        }
        key="68dda78f-364f-41dd-869a-0dd8d9a3fe4c"
        publishers={Object {}}
      />
      <SearchHitItem
        fadeInCounter={1}
        item={
          Object {
            "formats": Array [
              "application/xml",
              "application/json",
              "text/json",
              "text/xml",
            ],
            "id": "70f6e5b3-b633-413b-b5e7-11b63bb413eb",
            "nationalComponent": false,
            "publisher": Object {
              "id": "970018131",
              "name": "UTDANNINGSDIREKTORATET",
              "orgPath": "/STAT/872417842/970018131",
              "prefLabel": Object {
                "no": "Utdanningsdirektoratet",
              },
            },
            "title": "Nasjonalt barnehageregister API",
            "titleFormatted": "Nasjonalt barnehageregister API",
          }
        }
        key="70f6e5b3-b633-413b-b5e7-11b63bb413eb"
        publishers={Object {}}
      />
    </div>
  </div>
</main>
`;

exports[`should render ResultsApi correctly with minimum of props and no hits 1`] = `
<main
  data-test-id="apis"
  id="content"
>
  <div
    className="row mb-3"
  >
    <div
      className="col-6 col-lg-4"
    >
      <button
        className="btn btn-primary fdk-button fade-in-500 d-none"
        onClick={null}
        type="button"
      >
        Clear
      </button>
    </div>
    <div
      className="col-6 col-lg-4 offset-lg-4"
    >
      <div
        className="d-flex justify-content-end"
      >
        <Button
          className="fdk-button fdk-button-black-toggle selected"
          color="primary"
          onClick={[Function]}
          tag="button"
        >
          Relevance
        </Button>
        <Button
          className="fdk-button fdk-button-black-toggle"
          color="primary"
          onClick={[Function]}
          tag="button"
        >
          Recently published
        </Button>
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <aside
      className="search-filters col-lg-4 d-none d-lg-block"
    >
      <span
        aria-hidden="false"
        className="uu-invisible"
      >
        Filtrering
      </span>
    </aside>
    <div
      className="col-12 col-lg-8"
      id="apis"
    />
  </div>
</main>
`;
